compiledtrees/code_gen.py

```diff
@@ -153,26 +161,30 @@ def code_gen_ensemble(trees, individual_learner_weight, initial_value,
                     weight=individual_learner_weight)
                 gen.write(increment)
             gen.write("return result;")
-    return gen.lines
+    tree_files.append(gen.file)
+    return tree_files
+
+def _compile(cpp_f, tmpdir):
+    o_f = os.path.join(tmpdir, os.path.basename(cpp_f).replace('.cpp', '.o'))
+    _call([CXX_COMPILER, cpp_f, "-c", "-fPIC", "-o", o_f, "-O3"])
+    return o_f
 
+def _call(args):
+    DEVNULL = open(os.devnull, 'w')
+    subprocess.check_call(" ".join(args),
+                          shell=True, stdout=DEVNULL, stderr=DEVNULL)
 
-def compile_code_to_object(code):
+def compile_code_to_object(files):
     # XXX - should we clean up the temporary directory left from
     # mkdtemp()?
-    def call(args):
-        DEVNULL = open(os.devnull, 'w')
-        subprocess.check_call(" ".join(args),
-                              shell=True, stdout=DEVNULL, stderr=DEVNULL)
 
     tmpdir = tempfile.mkdtemp()
-    cpp_f = os.path.join(tmpdir, "tree.cpp")
     so_f = os.path.join(tmpdir, "tree.so")
-    o_f = os.path.join(tmpdir, "tree.o")
 
-    with open(cpp_f, 'w') as f:
-        f.write(code)
+    o_files = Parallel(n_jobs=-1, backend='threading')(delayed(_compile)(f, tmpdir) for f in files)
```

Comment:
Yes, exactly. Actually it's N+1. Additional .cpp file is the ensemble. 

-1 = all CPUs, -2 all but one, etc. I will expose it as n_jobs argument to the user, with default of 1. Im not the fan of hijacking the computer myself.